src/validation.cpp

```diff
@@ -2226,7 +2226,8 @@ bool CChainState::FlushStateToDisk(
     int nManualPruneHeight)
 {
     LOCK(cs_main);
-    assert(this->CanFlushToDisk());
+    const bool can_flush = this->CanFlushToDisk();
```

Comment:
Pretty sure this doesn't have side effects. Why does your tool think otherwise?

Reply:
You're right. Dropping this one.

Reply:
The function could be made const. Perhaps `cppcheck` then doesn't see this as potentially side-effect having?